File: adminpanel.Web/Modules/ecommercecom/category/categoryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace adminpanel.ecommercecom.Columns;

[ColumnsScript("ecommercecom.category")]
[BasedOnRow(typeof(categoryRow), CheckNames = true)]
public class categoryColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Photo { get; set; }
}

File: adminpanel.Web/Modules/ServerTypes/ecommercecom/categoryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { categoryRow } from "./categoryRow";

export interface categoryColumns {
    Id: Column<categoryRow>;
    Name: Column<categoryRow>;
    Description: Column<categoryRow>;
    Photo: Column<categoryRow>;
}

export class categoryColumns extends ColumnsBase<categoryRow> {
    static readonly columnsKey = 'ecommercecom.category';
    static readonly Fields = fieldsProxy<categoryColumns>();
}